### PR TITLE
LateInitUsage -> LateinitUsage

### LINKED ISSUE #513: test-pattern does not seem to be working
I've got the following configuration:

```yml
test-pattern: # Configure exclusions for test sources
  active: true
  patterns: # Test file regexes
    - '.*/test/.*'
    - '.*Test.kt'
    - '.*Spec.kt'
  exclude-rule-sets:
    - 'comments'
  exclude-rules:
    - 'NamingRules'
    - 'WildcardImport'
    - 'MagicNumber'
    - 'MaxLineLength'
    - 'LateInitUsage'
    - 'StringLiteralDuplication'
    - 'SpreadOperator'
    - 'TooManyFunctions'
    - 'LargeClass'
```

which is the default configuration and should be active on it's own, especially when used with `failFast`, right?

When I run `detektCheck` with RC5, I get the following report:

```
LateinitUsage - [keyboardHandler] at app/src/test/java/com/vanniktech/chessclock/android/keyboard/AndroidKeyboardHandlerTest.kt:27:3
```

which should not be reported since first of all the file naming matches and also the `test` directory. Could this be somehow related to #510 where the filepath would need to be resolved too?

### PR DESCRIPTION
**Pull request by vanniktech**
Should fix #513.

### CODE REVIEW

**Comment by vanniktech** [comment]
I'd rather prefer we fail on a mismatch.

File: CHANGELOG.md [hunk 1]
```diff
@@ -161,7 +161,7 @@ See all issues at: [RC2](https://github.com/arturbosch/detekt/milestone/16)
 - Update detekt-sample-ruleset - [#257](https://github.com/arturbosch/detekt/issues/257)
 - Fix input flag for standalone gradle task in README - [#256](https://github.com/arturbosch/detekt/pull/256)
 - Added rule for safe cast to config.yml - [#254](https://github.com/arturbosch/detekt/pull/254)
-- LateInitUsage: One annotation match is enough to let the property be ignored. - [#245](https://github.com/arturbosch/detekt/pull/245)
+- LateinitUsage: One annotation match is enough to let the property be ignored. - [#245](https://github.com/arturbosch/detekt/pull/245)
```
Comment (arturbosch) [inline CHANGELOG.md]: This is generated. As the issue had this description I would leave it.

File: CHANGELOG.md [hunk 2]
```diff
@@ -308,20 +308,20 @@ See all issues at: [M12](https://github.com/arturbosch/detekt/milestone/8)
 - SpacingAroundCurlyBraces throw IndexOutOfFound when determinating Location.of(node) as LineAndColumn calculation is simple wrong (of Idea?) - [#18](https://github.com/arturbosch/detekt/issues/18)
 
 
-See all issues at: [M11](https://github.com/arturbosch/detekt/milestone/5)
```
Comment (arturbosch) [inline CHANGELOG.md]: The two spaces at the end will force newline in markdown. Just don't change the Changelog.md please :)

Comment (vanniktech) [reply CHANGELOG.md]: Didn't know this. Interesting

**Comment by vanniktech** [comment]
> When we have a generating tool for the config this will be automated anyway?

That would be ideal.

**Pull request merged**